src/jit/regalloc.cpp

```diff
@@ -706,7 +706,10 @@ regNumber Compiler::raUpdateRegStateForArg(RegState* regState, LclVarDsc* argDsc
 
     if (regState->rsIsFloat)
     {
-        noway_assert(inArgMask & RBM_FLTARG_REGS);
+        if (!opts.compUseSoftFP)
+        {
+            noway_assert(inArgMask & RBM_FLTARG_REGS);
```

Comment:
This doesn't seem right. It seems like the caller should never call this function when both `regState->rsIsFloat` and `opts.compUseSoftFP` are true.